prometheus/metric.go

```diff
@@ -142,3 +145,31 @@ func NewInvalidMetric(desc *Desc, err error) Metric {
 func (m *invalidMetric) Desc() *Desc { return m.desc }
 
 func (m *invalidMetric) Write(*dto.Metric) error { return m.err }
+
+type timestampedMetric struct {
+	Metric
+	t time.Time
+}
+
+func (m timestampedMetric) Write(pb *dto.Metric) error {
+	e := m.Metric.Write(pb)
+	pb.TimestampMs = proto.Int64(m.t.Unix()*1000 + int64(m.t.Nanosecond()/1000000))
```

Comment:
Probably does not make a difference, but I’d return early in case of an error

Reply:
Somehow my comment here got lost.

It probably really doesn't make a difference, but I find it a bit more robust to set the timestamp even in case of an error. We don't explicitly document to not make any use of the protomessage after an error is returned, so there might be a future use where the error conveys some message that looking at the proto message still makes sense.

Right now, of course, nobody should ever look at the proto message after an error, but then it doesn't matter if we set the timestamp or not.